controllers/storagecluster/cephobjectstores.go

```diff
@@ -175,6 +186,16 @@ func (r *StorageClusterReconciler) newCephObjectStoreInstances(initData *ocsv1.S
 			r.Log.Error(err, "Failed to set ControllerReference for CephObjectStore.", "CephObjectStore", klog.KRef(obj.Namespace, obj.Name))
 			return nil, err
 		}
+		// if kmsConfig is not 'nil', add the KMS details to CephCluster spec
+		if kmsConfigMap != nil {
+			// Set default KMS_PROVIDER and VAULT_SECRET_ENGINE values, refer https://issues.redhat.com/browse/RHSTOR-1963
+			if _, ok := kmsConfigMap.Data["KMS_PROVIDER"]; !ok {
+				kmsConfigMap.Data["KMS_PROVIDER"] = "vault"
+			}
+			kmsConfigMap.Data["VAULT_SECRET_ENGINE"] = "transit"
```

Comment:
If you change `kmsConfigMap` it will change for the cluster-wide scenario too. So you need to populate a new map instead. If you don't on the next reconcile we will be passing `VAULT_SECRET_ENGINE = transit` to the OSD encryption scenario.
I'm not sure if the CephCluster and CephObjectStore CRs are reconciled later or not by OCS-Op but I'd like to avoid any issues with that. Also, I don't know if later in the code `kmsConfigMap` is reused or not.

So please create another instead of `kmsConfigMap` for the object store so we avoid any potential issues.

Reply:
Why is this line still needed? `kmsConfigMap` uses `kv` not `transit`.